api/http/handler/edgestacks/edgestack_status_update.go

```diff
@@ -84,6 +92,26 @@ func (handler *Handler) edgeStackStatusUpdate(w http.ResponseWriter, r *http.Req
 }
 
 func (handler *Handler) updateEdgeStackStatus(tx dataservices.DataStoreTx, r *http.Request, stackID portainer.EdgeStackID, payload updateStatusPayload) (*portainer.EdgeStack, error) {
+	stack, err := tx.EdgeStack().EdgeStack(stackID)
+	if err != nil {
```

Comment:
That's not the only possible error though, you should use `dataservices.IsErrObjectNotFound()` to ensure that's the case.